sklearn/utils/_metadata_requests.py

```diff
@@ -1251,6 +1254,56 @@ class _MetadataRequester:
     .. versionadded:: 1.3
     """
 
+    if TYPE_CHECKING:
```

Comment:
For coverage, I think we can mark the block as "nocover":

```suggestion
    if TYPE_CHECKING:  # pragma: no cover
```

XREF: https://coverage.readthedocs.io/en/stable/excluding.html#excluding-code-from-coverage-py